doc/reference/trd-appid.md

```diff
@@ -549,7 +549,7 @@ which credentials, are acceptable and which are rejected.
 
 ```rust
 pub enum CheckResult {
-    Accept,
+    Accept(Option<core::num::NonZeroUsize>),
```

Comment:
Can we try to have this be a generic type `T`?

That would avoid imposing overheads on credential checkers that don't need to pass addl. data, and makes this data unambiguous -- it can only be created by the checker and never be duplicated.

Reply:
I assume this is possible, but it is fairly invasive. `CheckResult<T>` the propagates to all of the checker machinery, which fine, but then because the `T` is stored in a `ProcessBinary` _that_ has to be templated on `T`, and at that point it is kind of everywhere, including code which isn't even using process checking.

Reply:
Yeah, I guess a `NonZeroUsize` is fine here... It's a bummer that it does increase the `CheckResult` size by another word.